packages/design-system/src/components/web-components/ds-drawer/ds-drawer.stories.tsx

```diff
@@ -0,0 +1,195 @@
+      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut
+      labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco
+      laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in
+      voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat
+      non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor
+      sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore
+      magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut
+      aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit
+      esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident,
+      sunt in culpa qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet,
+      consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna
+      aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex
+      ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum
+      dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa
+      qui officia deserunt mollit anim id est laborum. Lorem ipsum dolor sit amet, consectetur
+      adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad
+      minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo
+      consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu
+      fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui
+      officia deserunt mollit anim id est laborum.
+    </p>
+  </>
+);
+
+const Template = (args) => {
+  const [drawerOpen, setDrawerOpen] = useState(args['is-open'] ?? false);
+
+  useEffect(() => {
+    if (args['is-open'] !== drawerOpen) {
+      setDrawerOpen(args['is-open'] ?? false);
+    }
+  }, []);
+
+  useEffect(() => {
+    const drawerElement = document.querySelector('ds-drawer');
+    const toggleButton = document.querySelector('.ds-c-drawer__toggle');
+
+    if (drawerElement && toggleButton) {
+      const handleDrawerClose = (event: Event) => {
+        action('ds-close-click')(event);
+        setDrawerOpen(false);
+      };
+
+      const handleDrawerOpen = () => {
+        setDrawerOpen(true);
+      };
+
+      drawerElement.addEventListener('ds-close-click', handleDrawerClose as EventListener);
+      toggleButton.addEventListener('click', handleDrawerOpen);
+
+      return () => {
+        drawerElement.removeEventListener('ds-close-click', handleDrawerClose as EventListener);
+        toggleButton.removeEventListener('click', handleDrawerOpen);
+      };
+    }
+  }, [drawerOpen]);
```

Comment:
Tracking the state on this useEffect seems to resolve the above highlighted issue.